### PR TITLE
Adding class for Unit and Cells.

### PR DESCRIPTION
**Pull request by FlaminGuy**

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/jvm/GenCellClasses.scala [hunk 1]
```diff
@@ -13,15 +15,102 @@ object GenCellClasses {
     */
   def gen()(implicit root: Root, flix: Flix): Map[JvmName, JvmClass] = {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/GenCellClasses.scala]: Can we remove the Cell$ classes in this commit?

File: main/src/ca/uwaterloo/flix/language/phase/jvm/GenCellClasses.scala [hunk 2]
```diff
@@ -13,15 +15,102 @@ object GenCellClasses {
...
+
+    // Complete the visitor and get the bytecode.
+    visitor.visitEnd()
+    visitor.toByteArray
+  }
+
+  /**
+    * Generating constructor for the class `classType` with value of type `subType`
+    */
+  def genConstructor(classType: JvmType.Reference, subType: JvmType, visitor: ClassWriter)(implicit root: Root, flix: Flix): Unit = {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/GenCellClasses.scala]: subType is cellType?

File: main/src/ca/uwaterloo/flix/language/phase/jvm/GenUnitClass.scala [hunk 3]
```diff
@@ -12,8 +15,103 @@ object GenUnitClass {
     * Returns the bytecode for the Unit class built-in to the Flix language,
     */
   def gen()(implicit root: Root, flix: Flix): Map[JvmName, JvmClass] = {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/GenUnitClass.scala]: Same for removing the Unit class

File: main/src/ca/uwaterloo/flix/language/phase/jvm/JvmName.scala [hunk 4]
```diff
@@ -31,7 +31,8 @@ object JvmName {
   /**
     * The `ca.uwaterloo.flix.api.Unit` name
     */
-  val Unit: JvmName = JvmName(List("ca", "uwaterloo", "flix", "api"), "Unit")
+  // TODO: Perhaps a better directory?
+  val Unit: JvmName = JvmName(List("ca", "uwaterloo", "flix", "helps"), "Unit")
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/JvmName.scala]: Let us just put it into `flix.Unit`.

File: main/src/ca/uwaterloo/flix/language/phase/jvm/JvmName.scala [hunk 5]
```diff
@@ -129,6 +130,18 @@ object JvmName {
...
+
+  /**
+    * Get the class type for the cell with subtype `subType`
+    */
+  def getCellClassType(subType: JvmType): JvmType.Reference = {
+    val name = "Cell" + "$" + JvmOps.stringify(subType)
+
+    // TODO: Perhaps a better directory?
+    // The type resides in the ca.uwaterloo.flix.api.cell package.
+    JvmType.Reference(JvmName(List("ca", "uwaterloo", "flix", "helps", "cell"), name))
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/JvmName.scala]: `flix.Cell$X`

**Comment by magnus-madsen** [comment]
I added a few comments. Mainly I just want to ensure that this code works by removing the existing classes.

File: main/src/ca/uwaterloo/flix/language/phase/jvm/GenCellClasses.scala (hunk 1 above)
Comment (FlaminGuy) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/GenCellClasses.scala]: Fixed!

File: main/src/ca/uwaterloo/flix/language/phase/jvm/GenUnitClass.scala (hunk 3 above)
Comment (FlaminGuy) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/GenUnitClass.scala]: Fixed!

**Pull request merged**